packages/utils/package.json

```diff
@@ -78,6 +78,11 @@
     "rimraf": "*",
     "typescript": "*"
   },
+  "peerDependenciesMeta": {
+    "typescript": {
```

Comment:
[Question] Shouldn't this also be listed in `peerDependencies`? Even if listing it in `peerDependenciesMeta` implicitly adds it there, it feels odd to me to make that implicit.

Reply:
I just copied this from `type-utils`, which also does not list it as an explicit peerdep https://github.com/typescript-eslint/typescript-eslint/blob/42d275c183075be56166bb1704f96232f5d3cee5/packages/type-utils/package.json#L64-L68

Reply:
Ha, nice, so this might be an existing issue in that package then? My vote is to fix in both packages.

Reply:
The reason we used `peerDependenciesMeta` without `peerDependencies` was because there were versions of npm that did not support `peerDependenciesMeta`. This meant that when we used an explicit `peerDependencies` then users would be spammed with warnings saying they didn't have TS installed. This was a problem as some big packages included our plugins by default but only used them if the user opted in to TS which caused non-TS users to get spammed with missing peer dep errors.

A this point we could probably also include the explicit `peerDependencies` on TS but also there's no harm in not including it and having it inferred.

Reply:
Do you know if `npm` also shares this behavior of `peerDependenciesMeta` implying `peerDependencies`? From a cursory search of the npm sources it seems to me like it doesn't: https://github.com/npm/cli/blob/75a3f1228865f426d8790be27f1258e501f2c450/workspaces/arborist/lib/node.js#L859-L871

Anyways, what do we think of merging this PR first and moving the discussion about whether or not to list an explicit peerdep to a different issue? We don't need to block fixing users' warnings on figuring out the proper manifest declarations 😅

Reply:
I'm 👍 on that. If nobody else weighs in with a contrary opinion by our weekly release midday Monday EST then we can just merge as-is.

Now that you mention it, I do like the idea of having one release just adding `peerDependenciesMeta`, then a subsequent release adding `peerDependencies` on top. Just in case there's some bizarre edge case bug with the latter.